From 9327d45d0ef51ff82fb60539497e332f7133d33d Mon Sep 17 00:00:00 2001
From: Martin Jansa <martin2.jansa@lgepartner.com>
Date: Fri, 3 Jun 2022 19:19:57 +0000
Subject: [PATCH] Fix build with newer gcc

* fixes:
  http://gecko.lge.com/Errors/Details/415423

In file included from utils/CommonGraphOptions.h:28,
                 from utils/CommonGraphOptions.cpp:24:
./utils/command_line/CommandLineParser.h: In member function 'void arm_compute::utils::CommandLineParser::parse(int, char**)':
./utils/command_line/CommandLineParser.h:142:55: error: 'strlen' was not declared in this scope
  142 |         int         pos        = (equal_sign == -1) ? strlen(argv[i]) : equal_sign;
      |                                                       ^~~~~~
./utils/command_line/CommandLineParser.h:37:1: note: 'strlen' is defined in header '<cstring>'; did you forget to '#include <cstring>'?
   36 | #include <utility>
  +++ |+#include <cstring>
   37 | #include <vector>
In file included from utils/Utils.h:39,
                 from utils/Utils.cpp:24:
include/libnpy/npy.hpp: In function 'std::string npy::pyparse::write_tuple(const std::vector<T>&)':
include/libnpy/npy.hpp:298:42: error: 'ostream_iterator' is not a member of 'std'
  298 |     std::copy(v.begin(), v.end()-1, std::ostream_iterator<T>(ss, delimiter.c_str()));
      |                                          ^~~~~~~~~~~~~~~~
include/libnpy/npy.hpp:37:1: note: 'std::ostream_iterator' is defined in header '<iterator>'; did you forget to '#include <iterator>'?
   36 | #include <regex>
  +++ |+#include <iterator>
   37 | #include <unordered_map>
include/libnpy/npy.hpp:298:60: error: expected primary-expression before '>' token
  298 |     std::copy(v.begin(), v.end()-1, std::ostream_iterator<T>(ss, delimiter.c_str()));
      |                                                            ^
---
 include/libnpy/npy.hpp                 | 1 +
 utils/command_line/CommandLineParser.h | 1 +
 2 files changed, 2 insertions(+)

diff --git a/include/libnpy/npy.hpp b/include/libnpy/npy.hpp
index 24244ca27..7c9e5792f 100644
--- a/include/libnpy/npy.hpp
+++ b/include/libnpy/npy.hpp
@@ -27,6 +27,7 @@
 #include <fstream>
 #include <string>
 #include <iostream>
+#include <iterator>
 #include <sstream>
 #include <cstdint>
 #include <cstring>
diff --git a/utils/command_line/CommandLineParser.h b/utils/command_line/CommandLineParser.h
index e8fabc425..fa486b47a 100644
--- a/utils/command_line/CommandLineParser.h
+++ b/utils/command_line/CommandLineParser.h
@@ -27,6 +27,7 @@
 #include "Option.h"
 #include "arm_compute/core/utils/misc/Utility.h"
 
+#include <cstring>
 #include <iostream>
 #include <map>
 #include <memory>
